addons/event_booth_sale/models/sale_order_line.py

```diff
@@ -25,13 +26,20 @@ def _compute_name_short(self):
             record.name_short = record.event_booth_pending_ids.event_id.name
         super(SaleOrderLine, self - wbooth)._compute_name_short()
 
+    @api.constrains('event_booth_registration_ids')
+    def _check_event_booth_registration_ids(self):
+        if len(self.event_booth_registration_ids.mapped('event_booth_id.event_id')) > 1:
+            raise ValidationError(_('Registrations from the same Order Line must belong to a single event.'))
+
     @api.onchange('product_id')
-    def _onchange_booth_product_id(self):
+    def _onchange_product_id(self):
```

Comment:
Isn't it overriding existing method in event_sale ? Would name it like _onchange_product_id_booth, to avoi clashes with existing method